client/blocks/reader-suggested-follows/hooks/use-recommend-or-related-sites-query/index.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	FeedRecommendation,
+	useFeedRecommendationsQuery,
+} from 'calypso/data/reader/use-feed-recommendations-query';
+import { RelatedSite, useRelatedSites } from 'calypso/data/reader/use-related-sites';
+
+interface Author {
+	wpcom_login: string;
+	ID: string;
+	name: string;
+}
+
+interface QueryParams {
+	author?: Author;
+	siteId: number;
+	postId: number;
+}
+
+interface QueryOptions {
+	enabled: boolean;
+}
+
+const getResourceType = (
+	recommendedFeeds: FeedRecommendation[] | null | undefined,
+	relatedSites: RelatedSite[] | null | undefined
+) => {
+	if ( Array.isArray( recommendedFeeds ) && recommendedFeeds?.length > 0 ) {
+		return 'recommended';
+	}
+
+	if ( Array.isArray( relatedSites ) && relatedSites?.length > 0 ) {
+		return 'related';
+	}
+
+	return null;
+};
+
+export const useRecommendOrRelatedSitesQuery = ( query: QueryParams, options?: QueryOptions ) => {
+	const { author, siteId, postId } = query;
+	const userLogin = author?.wpcom_login || author?.ID;
+	const enabled = options?.enabled ?? true;
+
+	const {
+		data: recommendedFeeds,
+		isLoading: isLoadingRecommendedFeeds,
+		isSuccess: isSuccessRecommendedFeeds,
+	} = useFeedRecommendationsQuery( userLogin, {
+		enabled: !! userLogin && enabled,
+	} );
+
+	const hasRecommendedFeeds = recommendedFeeds && recommendedFeeds.length > 0;
+	const shouldLoadRelatedSites = isSuccessRecommendedFeeds && ! hasRecommendedFeeds;
```

Comment:
Do we need a check to make sure the modal is still open here, to prevent unnecessary requests? That would make it: if the recommended feeds request is complete AND there are no recommended feeds AND the modal is still open, fetch the related sites.

```suggestion
	const shouldLoadRelatedSites = enabled && isSuccessRecommendedFeeds && ! hasRecommendedFeeds;
```

Reply:
In theory, it is done [here](https://github.com/Automattic/wp-calypso/blob/e2e3b611e0c0ba7524aaa5354fe8c8f0704621b8/client/blocks/reader-suggested-follows/hooks/use-recommend-or-related-sites-query/index.ts#L48), but I agree with your suggestion.

Note: I saw now the page is making two requests to /read/site/${ siteId }/sites/related, but they are not related to the PR code. 

I think we need to have tasks to review duplicated requests on the pages and optimize it.